module_checksum_upgrade/tests/test_module.py

```diff
@@ -0,0 +1,132 @@
+            'Module directory checksum not computed properly',
+        )
+
+    def test_compute_checksum_dir_ignore_excluded(self):
+        """It should exclude .pyc/.pyo extensions from checksum
+        calculations"""
+        if not self.own_writeable:
+            self.skipTest("Own directory not writeable")
+        with tempfile.NamedTemporaryFile(suffix='.pyc', dir=self.own_dir_path):
+            self.assertEqual(
+                self.own_module._get_checksum_dir(), self.own_checksum,
+                'SHA1 checksum does not ignore excluded extensions',
+            )
+
+    def test_compute_checksum_dir_recomputes_when_file_added(self):
+        """It should return a different value when a non-.pyc/.pyo file is
+        added to the module directory"""
+        if not self.own_writeable:
+            self.skipTest("Own directory not writeable")
+        with tempfile.NamedTemporaryFile(suffix='.py', dir=self.own_dir_path):
+            self.assertNotEqual(
+                self.own_module._get_checksum_dir(), self.own_checksum,
+                'SHA1 checksum not recomputed',
+            )
+
+    def test_saved_checksums(self):
+        Imm = self.env['ir.module.module']
+        base_module = Imm.search([('name', '=', 'base')])
+        self.assertEqual(base_module.state, 'installed')
+        self.assertFalse(Imm._get_saved_checksums())
+        Imm._save_installed_checksums()
+        saved_checksums = Imm._get_saved_checksums()
+        self.assertTrue(saved_checksums)
+        self.assertTrue(saved_checksums['base'])
+
+    def test_get_modules_with_changed_checksum(self):
+        Imm = self.env['ir.module.module']
+        self.assertTrue(Imm._get_modules_with_changed_checksum())
+        Imm._save_installed_checksums()
+        self.assertFalse(Imm._get_modules_with_changed_checksum())
+
+
+@common.at_install(False)
+@common.post_install(True)
+class TestModuleAfterInstall(TransactionCase):
+
+    def setUp(self):
+        super(TestModuleAfterInstall, self).setUp()
+        self.own_module = self.env['ir.module.module'].search([
+            ('name', '=', MODULE_NAME),
+        ])
+
+    def test_get_modules_partially_installed(self):
+        Imm = self.env['ir.module.module']
+        self.assertTrue(
+            self.own_module not in Imm._get_modules_partially_installed())
+        self.own_module.button_upgrade()
+        self.assertTrue(
+            self.own_module in Imm._get_modules_partially_installed())
+        self.own_module.button_upgrade_cancel()
+        self.assertTrue(
+            self.own_module not in Imm._get_modules_partially_installed())
+
+    def test_upgrade_changed_checksum(self):
+        Imm = self.env['ir.module.module']
+        Bmu = self.env['base.module.upgrade']
+
+        # upgrade_changed_checksum commits, so mock that
+        self.env.cr.commit = mock.MagicMock()
+
+        Imm._save_checksums({})
+        changed_modules = Imm._get_modules_with_changed_checksum()
```

Comment:
You can change you own module checksum before calling this and then make sure it appears there. Pretty similar to https://github.com/OCA/server-tools/pull/1176/files#r172818652